### PR TITLE
ipv6 support

### PR DESCRIPTION
**Pull request by wenwenxiong**
### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note

```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kk/pkg/plugins/network/modules.go [hunk 1]
```diff
@@ -128,7 +129,8 @@ func deployCalico(d *DeployNetworkPluginModule) []task.Interface {
 			Template: templates.CalicoNew,
 			Dst:      filepath.Join(common.KubeConfigDir, templates.CalicoNew.Name()),
 			Data: util.Data{
-				"KubePodsCIDR":            d.KubeConf.Cluster.Network.KubePodsCIDR,
+				"KubePodsV4CIDR":            strings.Split(d.KubeConf.Cluster.Network.KubePodsCIDR, ",")[0],
+				"KubePodsV6CIDR":            strings.Split(d.KubeConf.Cluster.Network.KubePodsCIDR, ",")[1],
```
Comment (pixiake) [inline cmd/kk/pkg/plugins/network/modules.go:133]: Same as the InternalAddress, if the ipv6 address is not configured for the `KubePodsCIDR`, I think it should be reported `panic`.

File: cmd/kk/pkg/core/connector/host.go [hunk 2]
```diff
@@ -65,6 +66,14 @@ func (b *BaseHost) GetInternalAddress() string {
 	return b.InternalAddress
 }
 
+func (b *BaseHost) GetInternalIPv4Address() string {
+	return strings.Split(b.InternalAddress, ",")[0]
+}
+
+func (b *BaseHost) GetInternalIPv6Address() string {
```
Comment (pixiake) [inline cmd/kk/pkg/core/connector/host.go:73]: If the ipv6 address is not configured for the hosts, I think it should be reported `panic`.

File: cmd/kk/apis/kubekey/v1alpha2/cluster_types.go [hunk 3]
```diff
@@ -135,12 +135,15 @@ func (cfg *ClusterSpec) GenerateCertSANs() []string {
 		if host.Address != cfg.ControlPlaneEndpoint.Address {
 			extraCertSANs = append(extraCertSANs, host.Address)
 		}
-		if host.InternalAddress != host.Address && host.InternalAddress != cfg.ControlPlaneEndpoint.Address {
-			extraCertSANs = append(extraCertSANs, host.InternalAddress)
+		InternalIPv4Address := strings.Split(host.InternalAddress, ",")[0]
+		InternalIPv6Address := strings.Split(host.InternalAddress, ",")[1]
```
Comment (pixiake) [inline cmd/kk/apis/kubekey/v1alpha2/cluster_types.go:139]: If the ipv6 address is not configured for the hosts, Ithink it should be reported `panic`.

File: cmd/kk/pkg/core/connector/host.go [hunk 4]
```diff
@@ -65,6 +66,14 @@ func (b *BaseHost) GetInternalAddress() string {
...
+	return strings.Split(b.InternalAddress, ",")[1]
```
Comment (pixiake) [inline cmd/kk/pkg/core/connector/host.go:74]: Same as the above, if the ipv6 address is not configured for the hosts, I think it should be reported `panic`.

File: cmd/kk/pkg/plugins/network/modules.go (hunk 1 above)
Comment (pixiake) [inline cmd/kk/pkg/plugins/network/modules.go:133]: Same as the `InteranlAddress`, if the ipv6 address is not configured for the `KubePodsV6CIDR`, I think it should be reported `panic`.

**Comment by pixiake** [comment]
@wenwenxiong  Thanks for this contribution !  I had added some comments，I think it may be necessary to check the relevant parameters before obtaining the ipv6 address.

**Comment by wenwenxiong** [comment]
if user want to support IPV6 , the should config node ipv6 address, and write ipv6 address in config-same.yaml like i provider

File: docs/config-example.md [hunk 5]
```diff
@@ -5,9 +5,9 @@ metadata:
   name: sample
 spec:
```
Comment (wenwenxiong) [inline docs/config-example.md:6]: ipv6 address set in host internalAddresses

File: docs/config-example.md [hunk 6]
```diff
@@ -80,15 +80,15 @@ spec:
       ipipMode: Always  # IPIP Mode to use for the IPv4 POOL created at start up. If set to a value other than Never, vxlanMode should be set to "Never". [Always | CrossSubnet | Never] [Default: Always]
       vxlanMode: Never  # VXLAN Mode to use for the IPv4 POOL created at start up. If set to a value other than Never, ipipMode should be set to "Never". [Always | CrossSubnet | Never] [Default: Never]
       vethMTU: 0  # The maximum transmission unit (MTU) setting determines the largest packet size that can be transmitted through your network. By default, MTU is auto-detected. [Default: 0]
-    kubePodsCIDR: 10.233.64.0/18
-    kubeServiceCIDR: 10.233.0.0/18
+    kubePodsCIDR: 10.233.64.0/18,fc00::/48
```
Comment (wenwenxiong) [inline docs/config-example.md:83]: why user want to support ipv6, but not set ipv6 pods and service cird ? so if user relly want to support ipv6, the config must set with node ipv6 addresses and (pod,service)cidr.

**Pull request closed without merging**